internal/service/certificatesigningrequest.go

```diff
@@ -19,26 +19,27 @@ import (
 
 const DefaultEnrollmentCertExpirySeconds int32 = 60 * 60 * 24 * 7 // 7 days
 
-func signApprovedCertificateSigningRequest(ca *crypto.CA, request *api.CertificateSigningRequest) error {
+func signApprovedCertificateSigningRequest(ca *crypto.CA, request *api.CertificateSigningRequest) ([]byte, error) {
 	if request == nil {
-		return errors.New("signApprovedCertificateSigningRequest: CertificateSigningRequest is nil")
+		return nil, errors.New("signApprovedCertificateSigningRequest: CertificateSigningRequest is nil")
 	}
 
-	// NOTE: this currently only requires one approval; this will change in the future based on policy
-	if api.IsStatusConditionTrue(request.Status.Conditions, api.CertificateSigningRequestDenied) {
-		return fmt.Errorf("signApprovedCertificateSigningRequest: certificate signing request has been denied and cannot be signed")
-	}
-	if !api.IsStatusConditionTrue(request.Status.Conditions, api.CertificateSigningRequestApproved) {
-		return fmt.Errorf("signApprovedCertificateSigningRequest: certificate signing request is not approved and cannot be signed")
+	// TODO - we don't check for Approved status here anymore as it makes more sense to set Approved only after a certiicate
+	// is updated - otherwise a CSR may be marked as Approved but not actually make it through the entire Approval process (ex.
+	// certificate may be missing). this assumes the checks for "should be approved/signed" happen in the calling function
+	if request.Status.Conditions != nil && len(request.Status.Conditions) > 0 {
```

Comment:
1. ApproveCertificateSigningRequest() already checks if the CSR was denied - it looks like we're checking it twice
2. I don't understand why we shouldn't check if the CSR was already approved?

Reply:
1. Yes, that's true.
2. In this version approval is not set until after we know we have a certificate, so it comes later in the logic. We could do this differently, although that would require more time.

In short it may make sense to remove both checks

Reply:
It seems like I'm missing something.  If someone approves, and then approves again, is that not an error?

Reply:
You are right, we don't want to approve twice as this would overwrite the old cert. I'm putting a check in the POST approval ServiceHandler.